HW0/P5/P5.txt

```diff
@@ -1 +1,42 @@
 # Your answers here.
+#
+1. 255s;
+
+2. Each worker will now count only 256 / 8 = 31s for the first round, after which their are 8 numbers. Then, four of the workers can count the sum of these 8 numbers into 4 numbers using 1s. Similarly, two of the works can count the sum of these 4 numbers into 2 numbers using 1s. Finally, one of the workers will count the sum of these 2 numbers using 1s. Therefore, the total time used is (31 + 1 + 1 + 1) = 34s.
+
+3. It will take log(256) / log(2) = 8 seconds.
+
+4. It should take ceil(log(N) / log(2)) seconds.
+
+5. We need to find a critical point where it is less efficient to count all these bags on one's own than to distribute 2^N bags to two cashiers. The derivation will be:
+
+t1 = 2^N - 1 // count all these 2^N bags by one cashier
```

Comment:
I really like this analysis! It's much more elegant and precise than my solution.